checker/src/main/java/org/checkerframework/checker/signedness/SignednessAnnotatedTypeFactory.java

```diff
@@ -74,11 +73,73 @@ protected void addComputedTypeAnnotations(
         // but adding the top type to them, which permits flow-sensitive type refinement.
         // (When it is possible to default types based on their TypeKinds,
         // this whole method will no longer be needed.)
+        addSignednessGlbAnnotation(tree, type);
         addUnknownSignednessToSomeLocals(tree, type);
 
         super.addComputedTypeAnnotations(tree, type, iUseFlow);
     }
 
+    /**
+     * Refines the type of an expression such as array, method invocation or an integer primitive
+     * to @SignednessGlb if it is within the signed positive range. (i.e. its MSB is zero).
```

Comment:
The grammar is wrong here.  I think you can delete the first period on the line to fix it.

Reply:
You didn't do this.  I did so, but please address all comments (either by making a fix or by starting a conversation) rather than ignoring some.